client/peerset/src/lib.rs

```diff
@@ -798,7 +795,7 @@ mod tests {
 
 	fn next_message(mut peerset: Peerset) -> Result<(Message, Peerset), ()> {
 		let next = futures::executor::block_on_stream(&mut peerset).next();
-		let message = next.ok_or_else(|| ())?;
+		let message = next.ok_or(())?;
```

Comment:
```suggestion
		let message = next.unwrap_or(());
```

Reply:
This will return `Ok(((), peerset))` instead of `Err(())`

Reply:
let's not do that then :-)